mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSortOnSentDateTest.java

```diff
@@ -0,0 +1,218 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.lucene.search;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.mail.Flags;
+import javax.mail.Flags.Flag;
+
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.mock.MockMailboxSession;
+import org.apache.james.mailbox.model.MailboxACL;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.model.SearchQuery.Sort.SortClause;
+import org.apache.james.mailbox.model.SimpleMailboxACL;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.apache.james.mailbox.store.SimpleMailboxMembership;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.mail.model.Message;
+import org.apache.lucene.store.RAMDirectory;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LuceneMailboxMessageSortOnSentDateTest {
+
+    public static final long LIMIT = 100L;
+    private LuceneMessageSearchIndex index;
+    
+    private SimpleMailbox mailbox = new SimpleMailbox(0);
+    private MailboxSession session;
+
+    private MessageUid uid1;
+    private MessageUid uid2;
+    private MessageId id1;
+    private MessageId id2;
+
+    private Date smallestDate;
+    private Date biggestDate;
+
+    protected boolean useLenient() {
+        return true;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        session = new MockMailboxSession("username");
+        TestMessageId.Factory factory = new TestMessageId.Factory();
+
+        index = new LuceneMessageSearchIndex(null, new TestId.Factory(), new RAMDirectory(), true, useLenient(), factory);
+        index.setEnableSuffixMatch(true);
+
+        uid1 = MessageUid.of(1);
+        uid2 = MessageUid.of(2);
+
+        id1 = factory.generate();
+        id2 = factory.generate();
+
+        GregorianCalendar cal = new GregorianCalendar();
+
+        cal.set(2017, 0, 10);
+        smallestDate = cal.getTime();
+
+        cal.set(2017, 0, 12);
+        biggestDate = cal.getTime();
```

Comment:
Could not find it, I would be really happy if someone can suggest me an other way! (In Java 6)